File: lab2/task1.cpp
#include <iostream>
using namespace std;

int main() {
    int n = 0;
    int steps = 0;
    int sum = 0;
    cout << "Please enter a positive integer greater than 9:";
    cin >> n;

    while (n > 9) {
        if (n > 10) {
            cout << n << "-->";
        }
        else if (n == 10) {
            cout << 10 << "-->" << 1;
            steps++;
            break;
        }
        else {
            cout << n;
        }

        steps++;

        while (n > 0) {
            int temp = n %10;
            sum += temp;
            n /= 10;
        }

        n = sum;
        sum = 0;

    }

    cout << n;

    cout << endl << "Final Value:" << n;
    cout << endl << "Total Steps:" << steps;

    return 0;
}

File: lab2/task1.py
n = int(input("Please enter a positive integer greater than 9:"))
steps = 0
sum = 0
while n > 9:

    if n > 10:
        print(n , "→",end=" ")
    elif n == 10:
        print("10", "→", "1")
        steps+=1
        break
    else:
        print(n)
        break

    steps += 1

    while n > 0:
        temp = n
        temp %= 10
        sum += temp
        n //= 10

    n = sum
    sum = 0

print(n)
print("Final Value:", n)
print("Total Steps:", steps)